src/PointwiseFunctions/Hydro/InitialData/IrrotationalBns.cpp

```diff
@@ -0,0 +1,231 @@
+  return buffer;
+}
+
+template <typename DataType>
+void derivative_rotational_shift_over_lapse_squared(
+    const gsl::not_null<tnsr::iJ<DataType, 3>*> result,
+    const tnsr::I<DataType, 3>& rotational_shift,
+    const tnsr::iJ<DataType, 3>& deriv_of_shift, const Scalar<DataType>& lapse,
+    const tnsr::i<DataType, 3>& deriv_of_lapse,
+    const tnsr::iJ<DataType, 3>& deriv_of_spatial_rotational_killing_vector) {
+  ::tenex::evaluate<ti::i, ti::J>(
+      result, (deriv_of_shift(ti::i, ti::J) +
+               deriv_of_spatial_rotational_killing_vector(ti::i, ti::J)) /
+                      square(lapse()) -
+                  2.0 * rotational_shift(ti::J) * deriv_of_lapse(ti::i) /
+                      cube(lapse()));
+}
+
+template <typename DataType>
+tnsr::iJ<DataType, 3> derivative_rotational_shift_over_lapse_squared(
+    const tnsr::I<DataType, 3>& rotational_shift,
+    const tnsr::iJ<DataType, 3>& deriv_of_shift, const Scalar<DataType>& lapse,
+    const tnsr::i<DataType, 3>& deriv_of_lapse,
+    const tnsr::iJ<DataType, 3>& deriv_of_spatial_rotational_killing_vector) {
+  tnsr::iJ<DataType, 3> buffer{};
+  derivative_rotational_shift_over_lapse_squared(
+      make_not_null(&buffer), rotational_shift, deriv_of_shift, lapse,
+      deriv_of_lapse, deriv_of_spatial_rotational_killing_vector);
+  return buffer;
+}
+
+template <typename DataType>
+void enthalpy_density_squared(
+    const gsl::not_null<Scalar<DataType>*> result,
+    const tnsr::I<DataType, 3>& rotational_shift, const Scalar<DataType>& lapse,
+    const tnsr::i<DataType, 3>& velocity_potential_gradient,
+    const tnsr::II<DataType, 3>& inverse_spatial_metric,
+    const double euler_enthalpy_constant) {
+  return tenex::evaluate<>(
+      result,
+      square((euler_enthalpy_constant +
+              rotational_shift(ti::I) * velocity_potential_gradient(ti::i)) /
+             lapse()) -
+          velocity_potential_gradient(ti::i) *
+              velocity_potential_gradient(ti::j) *
+              inverse_spatial_metric(ti::I, ti::J));
+}
+
+template <typename DataType>
+Scalar<DataType> enthalpy_density_squared(
+    const tnsr::I<DataType, 3>& rotational_shift, const Scalar<DataType>& lapse,
+    const tnsr::i<DataType, 3>& velocity_potential_gradient,
+    const tnsr::II<DataType, 3>& inverse_spatial_metric,
+    const double euler_enthalpy_constant) {
+  Scalar<DataType> buffer{};
+  enthalpy_density_squared(make_not_null(&buffer), rotational_shift, lapse,
+                           velocity_potential_gradient, inverse_spatial_metric,
+                           euler_enthalpy_constant);
+  return buffer;
+}
+
+template <typename DataType>
+void spatial_rotational_killing_vector(
+    const gsl::not_null<tnsr::I<DataType, 3>*> result,
+    const tnsr::I<DataType, 3>& x,
+    const Scalar<DataType>& local_angular_velocity_around_z,
+    const Scalar<DataType>& sqrt_det_spatial_metric) {
+  // Cross product involves volume element in arbitrary coordinates
```

Comment:
What do you mean with arbitrary coordinates? Maybe just define the Levi-Civita tensor in the docs of this function.